urllib3/util/selectors.py

```diff
@@ -112,10 +117,9 @@ def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
                     if expires is not None:
                         current_time = monotonic()
                         if current_time > expires:
-                            raise OSError(errno=errno.ETIMEDOUT)
+                            raise OSError(errno.ETIMEDOUT, "Connection timed out")
                         if recalc_timeout:
-                            if "timeout" in kwargs:
-                                kwargs["timeout"] = expires - current_time
+                            kwargs["timeout"] = expires - current_time
```

Comment:
The old code was covering that (there's an intermediate commit in my fork that I was going to post for context and ask questions, but then figured out those answers myself).

All I changed at that snippet was to remove the check for `"timeout" in kwargs`, as that is now covered by the updated `recalc_timeout and "timeout" not in kwargs` check.